lib/ace/ext/papeeria/highlighter.coffee

```diff
@@ -71,13 +76,33 @@ define((require, exports, module) ->
             left: leftNearest
             right: rightNearest
             mismatch: true
-        if result.left and result.right
-            expectedRightBracket = session.$brackets[session.getLine(leftNearest.row).charAt(leftNearest.column)]
-            rightBracket = session.getLine(rightNearest.row).charAt(rightNearest.column)
+        closingBrackets = 
+            ")": "("
+            "]": "["
+            "}": "{"
+        openingBrackets = 
+            "(": ")"
+            "[": "]"
+            "{": "}"
+
+        # Next two 'if' needs to avoid conflict with the standard highlight
```

Comment:
It seems to me that you find it easier to see. Comment out line 79-96 and see what happends when you have case of {.....}_ or {_.....} (_ caret).